x-pack/plugins/cloud_security_posture/public/pages/findings/use_findings_count.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useQuery } from 'react-query';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type {
+  IKibanaSearchRequest,
+  IKibanaSearchResponse,
+} from '../../../../../../src/plugins/data/common';
+import { useKibana } from '../../common/hooks/use_kibana';
+import { showErrorToast } from './use_findings';
+import type { FindingsBaseQuery } from './findings_container';
+
+type FindingsAggRequest = IKibanaSearchRequest<estypes.SearchRequest>;
+type FindingsAggResponse = IKibanaSearchResponse<estypes.SearchResponse<{}, FindingsAggs>>;
+interface FindingsAggs extends estypes.AggregationsMultiBucketAggregateBase {
+  count: {
+    buckets: Array<{
+      key: string;
+      doc_count: number;
+    }>;
+  };
+}
+
+export const getFindingsCountAggQuery = ({ index, query }: FindingsBaseQuery) => ({
+  index,
+  size: 0,
+  track_total_hits: true,
+  body: {
+    query,
+    aggs: { count: { terms: { field: 'result.evaluation' } } },
+  },
+});
+
+export const useFindingsCounter = ({ index, query }: FindingsBaseQuery) => {
+  const {
+    data,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  return useQuery(
+    ['csp_findings_counts', { index, query }],
+    () =>
+      data.search
+        .search<FindingsAggRequest, FindingsAggResponse>({
+          params: getFindingsCountAggQuery({ index, query }),
+        })
+        .toPromise(),
+    {
+      onError: (error) => showErrorToast(toasts, error),
+      select: (response) =>
+        Object.fromEntries(
+          response.rawResponse.aggregations!.count.buckets.map((bucket) => [
+            bucket.key,
+            bucket.doc_count,
+          ])!
```

Comment:
looks like this can throw? maybe add error handling since you already have the hook

Reply:
it might, not likely tho. i could add a defensive check for it so the type wouldn't be `| undefined` 

in anyway, errors are already handled with react-query's `onError`